pkg/odo/cli/service/ui/ui.go

```diff
@@ -0,0 +1,307 @@
+	}
+	err := survey.AskOne(prompt, &plan, nil)
+	handleError(err)
+	return plan
+}
+
+// EnterServiceNameInteractively lets the user enter the name of the service instance to create, defaulting to the provided
+// default value and specifying both the prompt text and validation function for the name
+func EnterServiceNameInteractively(defaultValue, promptText string, validateName Validator) (serviceName string) {
+	// if only one arg is given, ask to Name the service providing the class Name as default
+	instancePrompt := &survey.Input{
+		Message: promptText,
+		Default: defaultValue,
+	}
+	err := survey.AskOne(instancePrompt, &serviceName, survey.Validator(validateName))
+	handleError(err)
+	return serviceName
+}
+
+// SelectClassInteractively lets the user select target service class from possible options, first filtering by categories then
+// by class name
+func SelectClassInteractively(classesByCategory map[string][]scv1beta1.ClusterServiceClass) (class scv1beta1.ClusterServiceClass, serviceType string) {
+	var category string
+	prompt := &survey.Select{
+		Message: "Which kind of service do you wish to create",
+		Options: getServiceClassesCategories(classesByCategory),
+	}
+	err := survey.AskOne(prompt, &category, survey.Required)
+	handleError(err)
+
+	classes := getServiceClassMap(classesByCategory[category])
+
+	// make a new displayClassInfo function available to survey templates to be able to add class information to the display
+	displayClassInfo := "displayClassInfo"
+	core.TemplateFuncs[displayClassInfo] = func(index int, pageEntries []string) string {
+		if index >= 0 && len(pageEntries) > index {
+			selected := pageEntries[index]
+			class := classes[selected]
+			return fmt.Sprintf("Name: %s\nDescription: %s\nLong: %s", class.GetExternalName(), class.GetDescription(), getLongDescription(class))
+		}
+		return "No matching entry"
+	}
+	defer delete(core.TemplateFuncs, displayClassInfo)
+
+	// record original template and defer restoring it once done
+	original := survey.SelectQuestionTemplate
+	defer restoreOriginalTemplate(original)
+
+	// add more information about the currently selected class
+	survey.SelectQuestionTemplate = original + `
+{{- if not .ShowAnswer}}
+{{$classInfo:=(displayClassInfo .SelectedIndex .PageEntries)}}
+  {{- if $classInfo}}
+===
+{{$classInfo}}
+  {{- end}}
+{{- end}}`
+
+	prompt = &survey.Select{
+		Message: "Which " + category + " service class should we use",
+		Options: getServiceClassNames(classes),
+	}
+
+	err = survey.AskOne(prompt, &serviceType, survey.Required)
+	handleError(err)
+
+	return classes[serviceType], serviceType
+}
+
+// restoreOriginalTemplate restores the original survey template once we're done with the display
+func restoreOriginalTemplate(original string) {
+	survey.SelectQuestionTemplate = original
+}
+
+// Convert the provided ClusterServiceClass to its UI representation
+func getLongDescription(class scv1beta1.ClusterServiceClass) (longDescription string) {
```

Comment:
I think some of these functions can be moved to some ui utils file or their core respective core folders. WDYT?

Reply:
Not sure yet, that's the purpose of #1126 but I'm not sure this function is generic enough unless we introduce a type / interface as I did for parameters.

Reply:
ok I guess #1126 will solve it